File: mashup-recruit/src/main/java/kr/mashup/branding/ui/applicant/vo/LoginResponse.java
package kr.mashup.branding.ui.applicant.vo;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;

@Data
public class LoginResponse {
    private final String accessToken;
    @JsonProperty("applicant")
    private final ApplicantResponse applicantResponse;

    public static LoginResponse of(String accessToken, ApplicantResponse applicantResponse){
        return new LoginResponse(accessToken, applicantResponse);
    }
}

File: mashup-recruit/src/main/java/kr/mashup/branding/ui/applicant/vo/LoginResponse.kt
package kr.mashup.branding.ui.applicant.vo

data class LoginResponse(
    val accessToken: String,
    val applicant: ApplicantResponse
) {

    companion object {
        @JvmStatic
        fun of(
            accessToken: String,
            applicantResponse: ApplicantResponse
        ): LoginResponse {
            return LoginResponse(accessToken, applicantResponse)
        }
    }
}